base/applications/taskmgr/procpage.c

```diff
@@ -1103,38 +1097,43 @@ static DWORD GetProcessExecutablePathById(DWORD dwProcessId, LPWSTR lpExePath, D
 void ProcessPage_OnProperties(void)
 {
     DWORD dwProcessId;
-    WCHAR szPath[MAX_PATH];
-    LPWSTR pszPath = NULL;
-    LPWSTR pszExePath = NULL;
+    LPWSTR pszExePath;
     DWORD dwLength;
     SHELLEXECUTEINFOW info = { 0 };
 
     dwProcessId = GetSelectedProcessId();
-    dwLength = GetProcessExecutablePathById(dwProcessId, szPath, _countof(szPath));
 
-    if (dwLength == 0)
+    /* Allocate a buffer for the path */
```

Comment:
Since in all cases (this call path, or the one below if `GetProcessExecutablePathById()` returns a longer length than MAX_PATH) you dynamically allocate a buffer, why not just more simply do one single call to `GetProcessExecutablePathById()` in order to retrieve that length ?